react-js/class-based-components/react-part-1-intro.md

```diff
@@ -0,0 +1,122 @@
+
+[Facebook](https://www.facebook.com) created React as a framework to create reuseable components, like posts or comments, and allow the user see updates instantly. [Instagram](https://www.instagram.com/) and [Airbnb](https://www.airbnb.com/) use it too.
+
+### Objectives
+
+**Participants will be able to:**
+
+- Explain the Pros and Cons of using React instead of other tools, like jQuery.
+- Create a simple React Component.
+
+### Specific Things to Learn
+
+- Why React?
+- What is JSX and why is it different than HTML?
+- Specifics about co-locating markup with logic in code.
+
+### Lesson Materials
+
+You will spend the day working through these materials. 
+
+1. [Introduction to React (video)](https://youtu.be/ycstRj2i66k)
+1. [Learn React in 5 minutes (article)](https://medium.freecodecamp.org/learn-react-js-in-5-minutes-526472d292f4)
+1. [Techtonica React slides](https://docs.google.com/presentation/d/1Bswkl7e1kGVav7KFabHrBgXhRi7mlINTbinxnUcGDy8/edit?usp=sharing)
+1. [Building your first component in React.js | Pluralsight (video)](https://youtu.be/K_jS1anlVAM)
+1. [Work through this React tutorial](https://facebook.github.io/react/index.html)
+1. [CodePen for Hello World Example](http://codepen.io/marcacyr/pen/NAyqgX)
+1. [CodePen Greeting with Input Example](http://codepen.io/marcacyr/pen/bZLVbj)
+1. [CodePen for Counter Example](http://codepen.io/marcacyr/pen/rLJVqR)
+1. [CodePen for List - Rendering using a loop in React](http://codepen.io/marcacyr/pen/KrQpYb)
+
+### Topics to learn & Facilitate
+
+- React lets you generate HTML using JavaScript
+- The HTML can change over time, unlike regular HTML
+- No jQuery needed
+
+### Examples
+
+#### CodePen - Hello World component
+
+In this CodePen, a basic Hello World component is rendered.  
+Try modifying line 5 of the JS file to see the changes made to the file rendered.  
+[CodePen of Example 1](https://codepen.io/jamesjose03/pen/ExVjYBM)
+
+#### CodePen - Component with a single prop
+
+Properties are values that are passed to the component for various uses. Here, the component has a prop called name.  
+Try modifying line 12 of the JS file by changing the value ABC and view the changes made.  
+[CodePen of Example 2](https://codepen.io/jamesjose03/pen/xxwGKvN)
+
+#### CodePen - Component assigning a `className`
+
+React uses className instead of class for CSS styling and other purposes.  
+Here, the component has a className heading. Try applying various styles for the component by modifying the CSS file.  
+[CodePen of Example 3](https://codepen.io/jamesjose03/pen/PoPqoqp)
+
+### Common Mistakes / Misconceptions
+
+1. In render functions, don't forget to return JSX. You can return null for an empty output.
+   Remove the return from a component in one of the CodePens and take note of the error message shown.
+2. Don't forget that expressions in JSX must be enclosed in brackets. Example: `<div>Hello {this.state.thing}</div>`
```

Comment:
In functional components it would just be `<div> Hello { variableName } <div>` right?